web/src/engine/websites/FlameComics.ts

```diff
@@ -31,17 +31,29 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
-        const blobMainImage = await Common.FetchImageAjax.call(this, page, priority, signal);
-        if (!page.Parameters?.secondaryPic) return blobMainImage;
-        const pageUrl = (page.Parameters.secondaryPic) as string;
-        const request = new Request(pageUrl, { headers: { Referer: this.URI.href } });
-        const response = await Fetch(request);
-        const b1 = await createImageBitmap(blobMainImage);
-        const b2 = await createImageBitmap(await response.blob());
-        return DeScramble(new ImageData(b1.width + b2.width, b1.height), async (_, ctx) => {
-            ctx.drawImage(b1, 0, 0);
-            ctx.drawImage(b2, b1.width, 0);
-        });
+        return !page.Parameters?.secondaryPic ? await Common.FetchImageAjax.call(this, page, priority, signal) :
+
+            this.imageTaskPool.Add(async () => {
+                if (!page.Parameters?.secondaryPic) return await this.FetchBlob(page.Link.href, signal);
+
+                const blobMainImage = await this.FetchBlob(page.Link.href, signal);
```

Comment:
Performing two downloads within a single task will affect the meaning of the rate limit of the queue, is this by intention?
e.g., when the queue is limited to **5** tasks per second, it means the queue may perform up to **10** downloads per second.

Reply:
Consider pushing two tasks instead (almost at the same time so they got close positions in the queue)
```typescript
const [ main, secondary ] = await Promise.all([ FetchBlob(), FetchBlob() ]);

async FetchBlob() {
    return this.imageTaskPool.Add(async () => {
        // fetch the image data and return blob
    });
}
```

Reply:
Oh, i see